collections/nemo_nlp/nemo_nlp/data/datasets/bert_pretraining.py

```diff
@@ -216,81 +203,75 @@ def match_target_seq_length(document, target_seq_length, filename,
+        b_document, b_line_idx = match_target_seq_length(
+            b_document, target_seq_length_b, b_filename, b_line_idx,
+            self.sentence_indices)
+
+        def truncate_seq_pair(a, b, max_num_tokens):
             # Truncates a pair of sequences to a maximum sequence length
-            while True:
-                total_length = len(a_document) + len(b_document)
-                if total_length <= max_num_tokens:
-                    break
+            while (len(a) + len(b)) > max_num_tokens:
+                # Truncate the longer sequence
+                if len(a) > len(b):
+                    trunc_document = a
+                else:
+                    trunc_document = b
 
-                trunc_document = a_document if len(
-                    a_document) > len(b_document) else b_document
                 if len(trunc_document) <= 1:
                     raise ValueError("Input text corpora probably too small. "
                                      "Failed to truncate sequence pair to "
-                                     "maximum sequence length.")
+                                     "maximum sequence legnth.")
 
+                # Randomly truncate from the front or the back
                 if random.random() < 0.5:
                     del trunc_document[0]
                 else:
                     trunc_document.pop()
 
         truncate_seq_pair(a_document, b_document, max_num_tokens)
 
-        output_ids = [self.tokenizer.special_tokens["[CLS]"]] + a_document + \
-                     [self.tokenizer.special_tokens["[SEP]"]] + b_document + \
-                     [self.tokenizer.special_tokens["[SEP]"]]
+        output_ids = [self.cls_id] + a_document + \
+                     [self.sep_id] + b_document + [self.sep_id]
 
         input_ids, output_mask = self.mask_ids(output_ids)
 
-        output_mask = np.array(output_mask, dtype=np.float32)
         input_mask = np.zeros(self.max_seq_length, dtype=np.float32)
         input_mask[:len(input_ids)] = 1
 
         input_type_ids = np.zeros(self.max_seq_length, dtype=np.int)
         input_type_ids[len(a_document) + 2:len(output_ids) + 1] = 1
 
-        while len(input_ids) < self.max_seq_length:
-            input_ids.append(0)
-            output_ids.append(0)
-            output_mask = np.append(output_mask, [0])
+        padding_length = max(0, self.max_seq_length - len(input_ids))
+        if padding_length > 0:
+            input_ids.extend([self.pad_id] * padding_length)
+            output_ids.extend([self.pad_id] * padding_length)
+            output_mask.extend([0] * padding_length)
 
         # TODO: wrap the return value with () for consistent style.
         return np.array(input_ids), input_type_ids,\
             np.array(input_mask, dtype=np.float32), np.array(output_ids),\
-            np.array(output_mask, dtype=np.float32), label
+            np.array(output_mask, dtype=np.float32), is_next
```

Comment:
a small style change to make it easier to read:
return (np.array(input_ids),
            input_type_ids,
            np.array(input_mask, dtype=np.float32),
            np.array(output_ids),
            np.array(input_mask, dtype=np.float32),
            np.array(output_ids),
            np.array(output_mask, dtype=np.float32),
            label,
            np.array(output_mask, dtype=np.float32),
            is_next)